src/core/style/expressions/ramp.js

```diff
@@ -10,20 +9,15 @@ export default class Ramp extends Expression {
      * @param {*} minKey Optional
      * @param {*} maxKey Optional
      */
-    constructor(input, palette, minKey, maxKey) {
-        input = implicitCast(input);
-        if (minKey !== undefined) {
-            minKey = implicitCast(minKey);
-            maxKey = implicitCast(maxKey);
+    constructor(input, palette, ...args) {
+        if (args.length > 0) {
```

Comment:
Any special reason for throwing an error on this validation? could we just ignore the extra parameters?

Reply:
The reason is that is easy to try to use the old functionality, and assume that things don't work as expected because there is a bug. However, I think we should do this kind of validation more often. It's much better to throw an error than to render something with unexpected behavior (relative to the user).